torch_xla/csrc/xla_sharding_util.h

```diff
@@ -80,11 +80,13 @@ class ShardingUtil {
   // based on the `sharding` spec. REPLICATED sharding should result in shards
   // identical to the input; OTHERS (tiled) sharding result in shards where
   // each data dimension is sharded across devices along the same dimension in
-  // the `tile_assignment`; the returned tensor shards vector is indexed by the
-  // device IDs. There is no data duplication. Shards are not padded in case the
-  // input tensor is not evenly partitionable, unless `padded` is set.
-  // The the returned tensors will be in 1:1 correspondence with the `devices`
-  // vector, so the `i`th result will belong on the `i`th device.
+  // the `tile_assignment`; MANUAL sharding result in shards where only the
+  // first device holds the full data; the returned tensor shards vector is
```

Comment:
> only the first device holds the full data

Is this by definition of manual sharding?

Reply:
This is not by definition, but by our implementation choice.

Reply:
@yeounoh Will that be replicated then?

Reply:
Per our offline discussion, abstain from manual sharding on input data.